src/libcollections/str.rs

```diff
@@ -1246,51 +1484,71 @@ impl str {
         core_str::StrExt::rmatch_indices(self, pat)
     }
 
-    /// Returns a `&str` with leading and trailing whitespace removed.
+    /// Returns a string slice with leading and trailing whitespace removed.
+    ///
+    /// 'Whitespace' is defined according to the terms of the Unicode Derived
+    /// Core Property `White_Space`.
     ///
     /// # Examples
     ///
+    /// Basic usage:
+    ///
     /// ```
     /// let s = " Hello\tworld\t";
-    /// assert_eq!(s.trim(), "Hello\tworld");
+    ///
+    /// assert_eq!("Hello\tworld", s.trim());
     /// ```
     #[stable(feature = "rust1", since = "1.0.0")]
     pub fn trim(&self) -> &str {
         UnicodeStr::trim(self)
     }
 
-    /// Returns a `&str` with leading whitespace removed.
+    /// Returns a string slice with leading whitespace removed.
+    ///
+    /// 'Whitespace' is defined according to the terms of the Unicode Derived
+    /// Core Property `White_Space`.
     ///
     /// # Examples
     ///
+    /// Basic usage:
+    ///
     /// ```
     /// let s = " Hello\tworld\t";
-    /// assert_eq!(s.trim_left(), "Hello\tworld\t");
+    ///
+    /// assert_eq!("Hello\tworld\t", s.trim_left());
     /// ```
     #[stable(feature = "rust1", since = "1.0.0")]
     pub fn trim_left(&self) -> &str {
         UnicodeStr::trim_left(self)
     }
 
-    /// Returns a `&str` with trailing whitespace removed.
+    /// Returns a string slice with trailing whitespace removed.
+    ///
+    /// 'Whitespace' is defined according to the terms of the Unicode Derived
+    /// Core Property `White_Space`.
     ///
     /// # Examples
     ///
+    /// Basic usage:
+    ///
     /// ```
     /// let s = " Hello\tworld\t";
-    /// assert_eq!(s.trim_right(), " Hello\tworld");
+    ///
+    /// assert_eq!(" Hello\tworld", s.trim_right());
     /// ```
     #[stable(feature = "rust1", since = "1.0.0")]
     pub fn trim_right(&self) -> &str {
         UnicodeStr::trim_right(self)
     }
 
-    /// Returns a string with all pre- and suffixes that match a pattern
-    /// repeatedly removed.
+    /// Returns a string slice with all prefixes and suffixes that match a
+    /// pattern repeatedly removed.
     ///
-    /// The pattern can be a simple `char`, or a closure that determines
+    /// The pattern can be a `&str`, [`char`], or a closure that determines
```

Comment:
@CamJN in the future, can you file a new issue rather than making a comment on an old PR? This comment will get lost in the sands of time.